test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/ObjectManagementServiceTests.cs

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.BatchParser.Utility;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Scripting;
+using Microsoft.SqlTools.ServiceLayer.Scripting.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Test.Common;
+using Moq;
+using NUnit.Framework;
+using static Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility.LiveConnectionHelper;
+
+namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.ObjectManagement
+{
+    public class ObjectManagementServiceTests
+    {
+        private readonly string table_query = @"CREATE TABLE testTable1_RenamingTable (c1 int)
+                            GO
+                            ";
+        private readonly string ownerUri = "testDB";
+        private readonly string startTableName = "testTable1_RenamingTable";
+        private readonly string startColumnName = "c1";
+        private ObjectManagementService objectManagementService;
+        private SqlTestDb testDb;
+        private Mock<RequestContext<bool>> requestContextMock;
+
+        [SetUp]
+        public async Task TestInitialize()
+        {
+            this.testDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, table_query, "RenameTest");
+
+            requestContextMock = new Mock<RequestContext<bool>>();
+            ConnectionService connectionService = LiveConnectionHelper.GetLiveTestConnectionService();
+
+            TestConnectionResult connectionResult = await LiveConnectionHelper.InitLiveConnectionInfoAsync(testDb.DatabaseName, this.ownerUri, ConnectionType.Default);
+
+            ObjectManagementService.ConnectionServiceInstance = connectionService;
+            this.objectManagementService = new ObjectManagementService();
+        }
+
+        [Test]
+        public async Task TestRenameTable()
+        {
+            await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenamingTable", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']", testDb.DatabaseName)), requestContextMock.Object);
+            Thread.Sleep(2000);
+            requestContextMock.Verify(x => x.SendResult(It.Is<bool>(r => VerifySendedFeedback(r, true))));
+            await CheckGeneratedScript(testDb.DatabaseName, "CREATE TABLE [dbo].[RenamingTable]");
```

Comment:
This seems more complex than I think it needs to be - you can just run the rename and verify that the old table name doesn't exist but the new one does with a direct query.

Reply:
Sorry, only one query. Do I understand that correctly, that I should execute a SQL-Query with the [OBJECT_ID-Function](https://learn.microsoft.com/en-us/sql/t-sql/functions/object-id-transact-sql?view=sql-server-ver16) from TSQL to prove if the item exists or even not?

Reply:
Something like that - yeah. (even just checking sys.tables should be enough but the general concept is the same)